packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -162,275 +109,408 @@ export const Block = Node.create<IBlock>({
+            ) {
+              return commands.BNMergeBlocks(posBetweenBlocks);
             }
+
             return false;
           }),
-        ({ chain }) =>
-          // we are at the start of a block at the root level. The user hits backspace to "merge it" to the end of the block above
-          //
-          // BlockA
-          // BlockB
-
-          // Becomes:
-
-          // BlockABlockB
-
-          chain()
-            .command(({ tr, state, dispatch }) => {
-              const isAtStartOfNode = tr.selection.$anchor.parentOffset === 0;
-              const anchor = tr.selection.$anchor;
-              const node = anchor.node(-1);
-              if (isAtStartOfNode && node.type.name === "block") {
-                if (node.childCount === 2) {
-                  // BlockB has children. We want to go from this:
-                  //
-                  // BlockA
-                  // BlockB
-                  //    BlockC
-                  //        BlockD
-                  //
-                  // to:
-                  //
-                  // BlockABlockB
-                  // BlockC
-                  //     BlockD
-
-                  // This parts moves the children of BlockB to the top level
-                  const startSecondChild = anchor.posAtIndex(1, -1) + 1; // start of blockgroup
-                  const endSecondChild = anchor.posAtIndex(2, -1) - 1;
-                  const range = state.doc
-                    .resolve(startSecondChild)
-                    .blockRange(state.doc.resolve(endSecondChild));
-
-                  if (dispatch) {
-                    tr.lift(range!, anchor.depth - 2);
-                  }
-                }
-                return true;
-              }
-              return false;
-            })
-            // use joinBackward to merge BlockB to BlockA (i.e.: turn it into BlockABlockB)
-            // The standard JoinBackward would break here, and would turn it into:
-            // BlockA
-            //     BlockB
-            //
-            // joinBackward has been patched with our custom version to fix this (see commands/joinBackward)
-            .joinBackward()
-            .run(),
-
-        () => commands.selectNodeBackward(), // (source: tiptap)
       ]);
 
     const handleEnter = () =>
       this.editor.commands.first(({ commands }) => [
-        // Try to split the current block into 2 items:
-        () => commands.splitListItem("block"),
-        // Otherwise, maybe we are in an empty list item. "Enter" should remove the list bullet
-        ({ tr, dispatch }) => {
-          const $from = tr.selection.$from;
-          if ($from.depth !== 3) {
-            // only needed at root level, at deeper levels it should be handled already by splitListItem
+        // Removes a level of nesting if the block is empty & indented, while the selection is also empty & at the start
+        // of the block.
+        () =>
+          commands.command(({ state }) => {
```

Comment:
can't we remove this (387-409), and use the code from before the change? `() => commands.splitListItem("block")`

Reply:
Yep, you're absolutely right, not sure why I overcomplicated this so much.

Reply:
Aha, the old implementation isn't compatible with the animations as it doesn't preserve block ID. Is that worth the extra complexity?